docs/fundamentals/networking/sockets/tcp-classes.md

```diff
@@ -78,10 +78,196 @@ The preceding C# code:
 - Writes the sent message to the console.
 - Finally, calls the <xref:System.Net.Sockets.TcpListener.Stop%2A> method to stop listening on the port.
 
+## Advanced TCP control with `Socket` class
+
+`TcpClient` and `TcpListener` relies on the `Socket` class internally, meaning you can achieve everything by using sockets directly. This guideline looks at several `TcpClient` and `TcpListener` use cases, presenting their `Socket` counterpart.
+
+### Create a client socket
+
+`TcpClient`'s default constructor tries to create a [_dual-stack socket_](<xref:System.Net.Sockets.Socket.DualMode>) via the [Socket(SocketType, ProtocolType)](xref:System.Net.Sockets.Socket.%23ctor%2A) constructor. This constructor creates a dual-stack socket if [IPv6 is supported](<xref:System.Net.Sockets.Socket.OSSupportsIPv6>), otherwise, it falls back to IPv4.
+
+Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient();
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+```
+
+#### The <xref:System.Net.Sockets.TcpClient.%23ctor(System.Net.Sockets.AddressFamily)?displayProperty=nameWithType> constructor
+
+This constructor accepts only three `AddressFamily` values, otherwise it will throw a <xref:System.ArgumentException>. Valid values are:
+
+- [AddressFamily.InterNetwork](xref:System.Net.Sockets.AddressFamily.InterNetwork): for IPv4 socket.
+- [AddressFamily.InterNetworkV6](xref:System.Net.Sockets.AddressFamily.InterNetworkV6): for IPv6 socket.
+- [AddressFamily.Unknown](xref:System.Net.Sockets.AddressFamily.Unknown): this will attempt to create a dual-stack socket, similarly to the default constructor.
+
+Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient(AddressFamily.InterNetwork);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+```
+
+#### The <xref:System.Net.Sockets.TcpClient.%23ctor(System.Net.IPEndPoint)?displayProperty=nameWithType> constructor
```

Comment:
Looks like there are no `displayProperty`-s other than `nameWithType` and `fullName`. What happens if you remove `?displayProperty=nameWithType` entirely? Would it simplify `TcpClient.TcpClient(String, Int32)` to `TcpClient(String, Int32)`?

Reply:
Let's try